org.mwc.debrief.legacy/src/Debrief/Tools/Tote/Calculations/DeltaRateToteCalcImplementation.java

```diff
@@ -0,0 +1,102 @@
+package Debrief.Tools.Tote.Calculations;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import MWC.GenericData.HiResDate;
+import junit.framework.TestCase;
+
+public class DeltaRateToteCalcImplementation {
+
+	public static class DeltaRateToteCalcImplementationTest extends TestCase {
+
+		public final static long TIME_WINDOW = 1000L * 60; // 60 seconds;
+
+		public void testCalculation() throws ParseException {
+			final String timeText = "22/04/09 17:59:58," + "22/04/09 18:00:06," + "22/04/09 18:00:14,"
+					+ "22/04/09 18:00:22," + "22/04/09 18:00:30," + "22/04/09 18:00:37," + "22/04/09 18:00:46,"
+					+ "22/04/09 18:00:54," + "22/04/09 18:01:02," + "22/04/09 18:01:10," + "22/04/09 18:01:18,"
+					+ "22/04/09 18:01:26," + "22/04/09 18:01:34";
+			final double[] measures = new double[] { 289.78, 289.61, 289.37, 289.18, 289, 289, 289.08, 289.38, 289.52,
+					289.6, 289.7, 289.73, 289.69 };
+			final double[] expectedAnswer = new double[] { 0, 0, 0.000226253464349, 0.000222222222222,
+					6.61375661376134E-05, -6.29154795821912E-05, 0, 0, -1.40996569567961E-05, 0.000298735655878,
+					0.000501339929911, 0.000499370118418, 0.000598702443941, };
+
+			final SimpleDateFormat format = new SimpleDateFormat("dd/MM/yy HH:mm:ss");
+
+			final String[] timeTokenized = timeText.split(",");
+			final Date[] timeDate = new Date[timeTokenized.length];
+			for (int i = 0; i < timeTokenized.length; i++) {
+				timeDate[i] = format.parse(timeTokenized[i]);
+			}
+
+			final HiResDate[] time = new HiResDate[timeDate.length];
+			for (int i = 0; i < time.length; i++) {
+				time[i] = new HiResDate(timeDate[i]);
+			}
+
+			final double[] deltaRateRate = DeltaRateToteCalcImplementation.calculate(measures, time, TIME_WINDOW);
+
+			assertArrayEquals("Delta Rate Tote Calculation Rate", expectedAnswer, deltaRateRate, 1e-5);
+		};
+	}
+
+	public static double[] calculate(final double[] measure, final HiResDate[] time, final long windowSizeMillis) {
+		if (measure.length > 2 && time.length == measure.length) {
+
+			// Let's calculate the delta rate
+			final double[] deltaRate = new double[measure.length];
```

Comment:
should this array be `length-1` - since we start at index 1. (since we're storing the intervals between the measurements)?